File: peterson.c
#include<stdio.h>
#include<conio.h>
int turn,x;
int  flag[2]={0,0};
int main()
{
     int i,j,n;
     printf("enter which process do you want you execute 0 or 1");
     scanf("%d",&n);
     if(n==0)
           goto p0;
     else 
          goto p1;
     p0:
			i=0;
			j=1;
			flag[i]=1;
	p0_start:
		turn=j;
		printf("p0::\n");
		while(flag[j]&&turn==j)
		{
			printf("#p0 in waiting state\n");
		
		   goto p1;
		   	
			}
			printf("#p0 critical section\n");
			  x=5;
			 flag[i]=0;
			
    
     p1:
			i=1;
			j=0;
			flag[i]=1;
	p1_start:
		turn=j;
		printf("#p1::\n");
		while(flag[j]&&turn==j)
		{  printf("#p1 is waiting state");
		   goto p0;
		   	
			}
				printf("#p1 critical section\n");
			 x=5;
			 flag[i]=0;
			 
			 return 0;
			
}

File: peterson.java
class process0 extends Thread
{
    public void run()
    {
      peterson.flag[0]="TRUE";
      peterson.turn=1;
     while(peterson.flag[1]=="TRUE" && peterson.turn==1)
    {
     System.out.println("P0 in blocked state.");
    }
    System.out.println("process0 is executing critical section");
    peterson.flag[0]="FALSE";
    }
}


class process1 extends Thread
{
   public void run()
  {
    peterson.flag[1]="TRUE";
    peterson.turn=0;
    while(peterson.flag[0]=="TRUE" && peterson.turn==0)
    {
    System.out.println("P1 in blocked state.");
    }
    System.out.println("process1 is executing critical section");
    peterson.flag[1]="FALSE";
    }
}


class peterson 
{
   static String flag[]={"FALSE","FALSE"};
   public static int turn;
   public static void main(String ar[])
  {
  process0 ob=new process0();
  process1 ob1=new process1();
  ob.start();
  ob1.start();
  }
}